### PR TITLE
Fix react/jsx key eslint errors

### PR DESCRIPTION
**Pull request by BCerki**
This PR:

- sets the react/jsx key rules to "error" instead of "off"
- fixes the react/jsx key eslint lint errors
- partially addresses Fixing eslint rule errors Fixing eslint rule errors Fixing eslint rule errors #37

### CODE REVIEW

File: src/front-end/typescript/lib/pages/proposal/sprint-with-us/view/tab/code-challenge.tsx [hunk 1]
```diff
@@ -170,7 +170,7 @@ const Participants: View<Pick<State, 'proposal'>> = ({ proposal }) => {
             {participants.map((p, i) => {
               const username = p.idpUsername;
               return (
-                <tr>
+                <tr key={username}>
```
Comment (BCerki) [inline src/front-end/typescript/lib/pages/proposal/sprint-with-us/view/tab/code-challenge.tsx]: Not 100% confident about this--normally I'd put the key on the component tag, but there isn't one here

Comment (wenzowski) [reply src/front-end/typescript/lib/pages/proposal/sprint-with-us/view/tab/code-challenge.tsx]: Right place for the key; any reason not to use `i`? Wouldn't that be guaranteed to be unique?

```suggestion
                <tr key={`participant-${i}`}>
```

Comment (mikevespi) [reply src/front-end/typescript/lib/pages/proposal/sprint-with-us/view/tab/code-challenge.tsx]: I second using `i` here

**Comment by BCerki** [comment]
pre-commit.ci run

**Review by mikevespi — APPROVED** [review]
Awesome! 🚢

**Pull request merged**